store/database.go

```diff
@@ -68,11 +68,11 @@ type VersionedDatabase interface {
 // Committer defines an API for committing state.
 type Committer interface {
 	WriteBatch(cs *Changeset) error
-	WorkingHash() []byte
-	GetLatestVersion() uint64
+	WorkingStoreInfos(version uint64) []StoreInfo
+	GetLatestVersion() (uint64, error)
 	LoadVersion(targetVersion uint64) error
-	Commit() ([]byte, error)
-	GetProof(version uint64, key []byte) (*ics23.CommitmentProof, error)
+	Commit() ([]StoreInfo, error)
```

Comment:
does this change the logic from returning an apphash to returning many apphashes? what is the benefit to this?